src/IdOnObjectSelectionSetRule.ts

```diff
@@ -68,3 +82,18 @@ function selectionSetIncludesId(selectionNodes: readonly SelectionNode[]) {
     }
   });
 }
+
+/**
+ * Look for the FooDetail/code/name pattern of our enum detail objects.
+ *
+ * NOTE: Borrowed from graphql-typescript-factories. A helper library may be in order?
+ */
+function isEnumDetailObject(object: GraphQLNamedType): object is GraphQLObjectType {
+  return (
+    object instanceof GraphQLObjectType &&
+    object.name.endsWith("Detail") &&
```

Comment:
Are we ensuring somewhere that all of our enum types end with "Detail"?

Reply:
Yeah, it's part of the joist codegen infra on the `graphql-service` side of things.

Reply:
Yes, we need to move this helper into it's own repo by now 😆